### PR TITLE
Feature / Full Tile

### PR DESCRIPTION
**Pull request by joebell93**
**Dependant on https://github.com/sky-uk/toolkit-core/pull/115**

Full Tile styles.

## Description
`.c-tile--full` for tiles that use images without a split. These tiles are white and use a new gradient on hover. `.has-focus` state for keyboard navigation.

## Related Issue
https://github.com/sky-uk/toolkit-ui/issues/128

## Motivation and Context
Cleaner design and accessibility improvements using `.has-focus`.

## How Has This Been Tested?
Tests pass on `npm run test`.
Visually checked in all browsers.

## Screenshots (if appropriate):
Achieves the following styles:
screen shot 2016-09-12 at 11 33 59

## Types of changes
What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] ~~Bug fix (non-breaking change which fixes an issue)~~
- [x] New feature (non-breaking change which adds functionality)
- [x] ~~Breaking change (fix or feature that would cause existing functionality to change)~~

## Checklist:

- [x] New functions and mixins have appropriate tests.
- [x] All new and existing tests passed.
- [x] Changes have been browser tested (including IE).
- [x] I have added instructions on how to test my changes.

### CODE REVIEW

**Comment by joebell93** [comment]
Needs to be reviewed by @danieljbryson before continuing

File: components/_tile.scss [hunk 1]
```diff
@@ -142,6 +148,14 @@ $included-tile-themes: "sky-1", "sky-account", "sky-atlantic", "sky-arts", "sky-
     margin-bottom: 0;
   }
 
+  /**
+   * Overlap and stretch body to fit full height on full tiles
+   */
+  .c-tile--full & {
```
Comment (StefanMcCready) [inline components/_tile.scss]: Still not convinced about the modifier. Don't know how anyone else feels?

Comment (MrDinsdale) [inline components/_tile.scss]: I understand the concern however this does align with design and existing templates on sky-pages. I've tried to think of a more descriptive modifier name but struggled!

**Comment by MrDinsdale** [comment]
Assuming design sign-off then its a :shipit:

File: components/_tile.scss [hunk 2]
```diff
@@ -47,6 +49,10 @@ $included-tile-themes: "sky-1", "sky-account", "sky-atlantic", "sky-arts", "sky-
   will-change: transform;
   z-index: 10; /* [1] */
 
+  &.has-focus {
```
Comment (aaronthomas) [inline components/_tile.scss]: All of our stateful classes are prefixed with `.is-`, this one seems to be breaking convention. `.is-focussed` perhaps?

Comment (MrDinsdale) [inline components/_tile.scss]: `.is-` and `.has-` are both listed in the [FAQs](https://github.com/sky-uk/toolkit/wiki/FAQ) however we haven't used `.has-` to my knowledge yet. I'd personally be fine with either.

Comment (joebell93) [inline components/_tile.scss]: I don't really mind either, `has-focus` felt more grammatically appropriate.

Shouldn't be too new to developers either with js like `document.hasFocus();`

File: components/_tile.scss (hunk 1 above)
Comment (joebell93) [inline components/_tile.scss]: Unless we can think of a more appropriate name, it's gonna have to be `--full` for now

**Comment by skitson** [comment]
:shipit:

File: components/_tile.scss (hunk 2 above)
Comment (aaronthomas) [inline components/_tile.scss]: Makes sense – feels odd to have two different naming conventions for the same thing, but agreed it's a practical name. Happy to stick with this.

:shipit:

**Pull request merged**